Warn when you make missing attribute searchable

ellnix opened the pull request:
# Pull Request

## Related issue
Fixes #168

## What does this PR do?
- In addition to adding a warning as suggested in #168 it also adds a `#logger` method to `MeiliSearch::Rails` which is similar to Rails having its own `Rails#logger` 

## PR checklist
Please check if your PR fulfills the following requirements:
- [X] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [X] Have you read the contributing guidelines?
- [X] Have you made sure that the title is accurate and descriptive of the changes?

brunoocasali commented on lib/meilisearch-rails.rb:
```diff
@@ -444,6 +448,9 @@ def meilisearch(options = {}, &block)
             after_destroy { |searchable| searchable.ms_enqueue_remove_from_index!(ms_synchronous?) }
           end
         end
+
```
```suggestion
```

brunoocasali commented on lib/meilisearch-rails.rb:
```diff
@@ -858,6 +865,17 @@ def ms_attribute_changed?(document, attr_name)
       end
+
+      def warn_searchable_missing_attributes
+        if (searchables = meilisearch_settings.get_setting(:searchable_attributes)) &&
```
It is okay to take some extra new lines to create the variables first. This way it will be easier to understand the logic behind the alert.

brunoocasali commented on spec/integration_spec.rb:
```diff
@@ -1036,6 +1036,25 @@
 
+context 'when a searchable attribute is not an attribute' do
+  let(:others) { Class.new(People) { def self.name = "People" } }
+
```
```suggestion
```

brunoocasali commented on spec/integration_spec.rb:
```diff
@@ -1036,6 +1036,25 @@
 end
 
+context 'when a searchable attribute is not an attribute' do
+  let(:others) { Class.new(People) { def self.name = "People" } }
```
Call it `people_class` or something like that, it will be easier to read afterwards.

brunoocasali commented on spec/integration_spec.rb:
```diff
@@ -1036,6 +1036,25 @@
+    allow(MeiliSearch::Rails).to receive(:logger).and_return(logger)
+
+    others.meilisearch index_uid: safe_index_uid('Others'), primary_key: :card_number do
+        attribute :first_name
+        searchable_attributes [:first_name, :last_name]
```
```suggestion
      attribute :first_name
      searchable_attributes [:first_name, :last_name]
```

brunoocasali commented on lib/meilisearch-rails.rb:
```diff
@@ -39,6 +39,10 @@ def included(klass)
       end
+
+      def logger
+        @logger ||= (::Rails.logger || Logger.new($stdout))
```
Nice catch!

brunoocasali reviewed (changes requested):
Thanks a lot @ellnix, for all of your work here!

I made some comments I would like you to reevaluate before we can merge this.

ellnix commented:
I believe I fixed the issues suggested, I had made some really strange decisions on this PR.

I also removed one place where I used the new ruby 3 method definition syntax which would have broken compatibility with ruby 2.x in the test suite.

brunoocasali commented:
> I believe I fixed the issues suggested, I had made some really strange decisions on this PR.
> 
> I also removed one place where I used the new ruby 3 method definition syntax which would have broken compatibility with ruby 2.x in the test suite.

Nice, as soon you fix them we will be good to go, thanks a lot for your effort @ellnix!

ellnix commented:
@brunoocasali this should be good to go.

brunoocasali reviewed (approved):
@ellnix thanks for this fix 🎉 

Thanks a lot for your patience and your smart work here ❤️

ellnix commented:
bors merge
